presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -62,14 +64,17 @@
     // for each source, the list of inputs corresponding to each output
     private final List<List<Symbol>> inputs;
 
+    private final Optional<OrderingScheme> orderingScheme;
+
```

Comment:
Yeah, it is weird. However, `ExchangeNode` and `ExchangeOperator` definitely do not preserve order -- they just return the pages as they come from all of the remote sources. We use this `ExchangeNode` to create a remote source that knows about ordering so you can merge. We do sort before the exchange, but you additionally need to merge the inputs as they come in. It's best to do this in the exchange because you need to be able to map pages to where they're coming from (for the merge sort algorithm -- since you need to have a page from all of the sources so you are sure that you are taking the smallest value).